document/core/valid/instructions.rst

```diff
@@ -480,83 +520,43 @@ Memory Instructions
    }
 
 
-.. _valid-memory.fill:
+.. _valid-memory.init:
 
-:math:`\MEMORYFILL`
+:math:`\MEMORYINIT~x`
 .....................
 
 * The memory :math:`C.\CMEMS[0]` must be defined in the context.
 
-* Then the instruction is valid with type :math:`[\I32~\I32~\I32] \to []`.
-
-.. math::
-   \frac{
-     C.\CMEMS[0] = \memtype
-   }{
-     C \vdashinstr \MEMORYFILL : [\I32~\I32~\I32] \to []
-   }
-
-
-.. index:: table instruction, table index, context
-   pair: validation; instruction
-   single: abstract syntax; instruction
-.. _valid-instr-table:
-
-Table Instructions
-~~~~~~~~~~~~~~~~~~
-
-.. _valid-table.init:
-
-:math:`\TABLEINIT~x`
-.....................
-
-* The table :math:`C.\CTABLES[0]` must be defined in the context.
-
-* The element segment :math:`C.\CELEM[x]` must be defined in the context.
+* The data segment :math:`C.\CDATA[x]` must be defined in the context.
 
 * Then the instruction is valid with type :math:`[\I32~\I32~\I32] \to []`.
 
 .. math::
    \frac{
-     C.\CTABLES[0] = \tabletype
+     C.\CMEMS[0] = \memtype
      \qquad
-     C.\CELEM[x] = \segtype
+     C.\CDATA[x] = \segtype
    }{
-     C \vdashinstr \TABLEINIT~x : [\I32~\I32~\I32] \to []
+     C \vdashinstr \MEMORYINIT~x : [\I32~\I32~\I32] \to []
    }
 
 
-.. _valid-table.drop:
+.. _valid-data.drop:
 
-:math:`\TABLEDROP~x`
-.....................
-
-* The element segment :math:`C.\CELEM[x]` must be defined in the context.
-
-* Then the instruction is valid with type :math:`[] \to []`.
-
-.. math::
-   \frac{
-     C.\CELEM[x] = \segtype
-   }{
-     C \vdashinstr \TABLEDROP~x : [] \to []
-   }
-
-
-.. _valid-table.copy:
+:math:`\DATADROP~x`
+...................
 
-:math:`\TABLECOPY`
-.....................
+* The data segment :math:`C.\CDATA[x]` must be defined in the context.
 
-* The table :math:`C.\CTABLES[0]` must be defined in the context.
+* The :ref:`segment type <syntax-segtype>` :math:`C.\CDATA[x]` must be |SPASSIVE|.
```

Comment:
This isn't correct anymore; not sure if you want to remove in this PR, since it's mostly just moving things around.

Reply:
Right, it's probably better to separate such fixes. I'll create an issue to track.